seastar/retrieval/level2.py

```
@@ -251,3 +258,146 @@ def compute_current_magnitude_and_direction(level1, level2):
+                              "method" : "wind",      # Can be "wind", "current", or "windcurrent"
+                              "truth" : geo}  
+            Defaults to None.
+        dict_env : ``dict``, (optional)
+            Dictionary containing the environnement information needed for SCR inversion. Shall contain 'u10' and 'wind_direction'.
+            Defaults to None.
+        write_nc : bool (optional)
+            Argument to write the data in a netcdf file. Defaults to False.
+        L1_folder : ``str``, (optional)
+            Path to save the L2 OSCAR data. Defaults to ".".
+
+    Returns:
+    ----------
+        ds_L2: ``xr.Dataset``
+            Xarray dataset of the L2 OSCAR data.
+    """
+
+    # Checking doppler GMF name: TODO: dev a function that test the entries
+    gmf_dict = dict_L2_process['gmf']
+    seastar.oscar.tools.is_valid_gmf_dict(gmf_dict)         # Check the format of gmf_dict
+    
+    #-----------------------------------------------------------
+    #               L2 PROCESSING
+    #-----------------------------------------------------------
+    # L2_processor, set by default on SCR (Sequential Current Retrieval) 
+    L2_processor = dict_L2_process.get("L2_processor","SCR")
+    
+    if L2_processor == "SCR":
+        logger.info(f"The processor chosen is: {L2_processor}")
+        print(dict_env)
+        
+        ds_L2 = seastar.retrieval.level2.sequential_current_retrieval(ds_L1, dict_env, gmf_dict['doppler']['name']) 
+        
+    elif L2_processor == "WCR":
+        logger.info(f"The processor chosen is: {L2_processor}")
+        ds_L1 = ds_L1.load()
+        
+        logger.info("Compute uncertainty")
+        uncertainty = xr.Dataset({"RSV":ds_L1.RadialSurfaceVelocity.copy(deep=True),
+                            "Kp":ds_L1.Intensity.copy(deep=True)})
+        uncertainty["RSV"] = dict_L2_process.get("RSV_Noise", 0.1)          # RSV_Noise, set by default at 0.1 
```

The defaults, and whether we should have default noise constants at all, should be debated I think. At the very least a warning should be raised to alert us, otherwise personally I think we should be inputting Kp and RSV_noise as a matter of course.

We also, potentially, may want to introduce variable uncertainty parameters (as in, images). But this can come later.

---

I agree, the noise should be Required as an input.

---

The RSV_noise and Kp are reported in the  dict_L2_process dict:
RSV_noise = 0.2
Kp = 0.2
dict_L2_process = {"gmf":gmf_dict, "L2_processor" : "SCR", "RSV_Noise": RSV_noise, "Kp" : Kp}

dans the line uncertainty["RSV"] = dict_L2_process.get("RSV_Noise", 0.1) check if RSV_noise exists in the dict and if not take 0.1 as default.
If it is too confusing to have a big Lé process dinctionnairy I can define a dict_uncertainty with only RSV_noise and Kp inside.

---

I believe it is ok to have a big "dict_L2_process", but I am not in favour to put a default value. If there is no value, I would raise an exception